src/update-git-refs.ts

```diff
@@ -79,6 +80,31 @@ function findTrackables(doc: yaml.Document.Parsed): Trackable[] {
       'ref',
     );
 
+    let maybeDockerCommit: string | null = null;
+
+    // We only care about using the potential docker commit sha if we are using the global tracker
+    // as opposed to defining a tracker for the git config by itself. We don't need to care if no track is present because we will not update those anyway
+    if (
+      value.has('dockerImage') &&
+      !getStringValue(gitConfigBlock, 'trackMutableRef')
+    ) {
+      const dockerImageBlock = value.get('dockerImage');
+      if (!yaml.isMap(dockerImageBlock)) {
+        throw Error(`Document has \`${key}.dockerImage\` that is not a map`);
+      }
+
+      const tagScalarTokenAndValue = getStringAndScalarTokenFromMap(
```

Comment:
do you need to use the "and scalar token" version? That's just for when we write back.  getStringValue should be fine right?

also for consistency we can consider reading a globalDockerCommit from global like we read globalRepoURL etc...